zerver/lib/cache.py

```diff
@@ -326,7 +326,8 @@ def user_profile_by_api_key_cache_key(api_key: str) -> str:
     'id', 'full_name', 'short_name', 'email',
     'avatar_source', 'avatar_version', 'is_active',
     'is_realm_admin', 'is_bot', 'realm_id', 'timezone',
-    'date_joined', 'is_guest', 'bot_owner_id'
+    'date_joined', 'is_guest', 'bot_owner_id',
+    'delivery_email'
```

Comment:
I was worried that this change will end up in the `delivery_email` being included in some data structures where it shouldn't (e.g. the `GET /users` API call).  But I think this is OK; we should just add a comment on this to make it clear the results of queries against this object shouldn't be sent to users.

(And maybe a test that `GET /users` and `/register` don't include a `delivery_email` field, if we don't have one already).

Reply:
Ohh, but `get_members_backend` don't use `get_realm_user_dicts` to fetch user information.